### ISSUE TITLE
Context does not get passed through correctly when calling a class function via `.instance()`

### ISSUE THREAD

**Issue by connorjburton** [mjancarik/shallow-with-context#2]
See below for code example. `this.context` in `foo` ends up looking like `Object {value: Object, __SHALLOW_CONTEXT_FLAG__: true}`

```js
import { shallow } from 'enzyme';
import { withContext, createContext } from 'shallow-with-context';
import React from 'react';

const MyContext = React.createContext({ text: 'default' });
class Component extends React.Component {
  static contextType = MyContext;
  foo() {
      return this.context.text;
  }

  render() {
      return <div>{this.context.text}</div>;
  }
}

describe('your description', () => {
    it('your spec', () => {
        const context = createContext({ text: 'new value' });
        const ComponentWithContext = withContext(Component, context);

        const wrapper = shallow(<ComponentWithContext />, { context });

        expect(wrapper.instance().foo()).toBe('new value'); // fails as `.foo()` returns `undefined`
    });
});
```

**Comment by mjancarik** [comment]
Hi, 

Thanks for report. Workaround than I fix it in the next version. You can pass your object value context direct to withContext method.

``` jsx
import { shallow } from 'enzyme';
import { withContext } from 'shallow-with-context';
import React from 'react';

const MyContext = React.createContext({ text: 'default' });
class Component extends React.Component {
  static contextType = MyContext;
  foo() {
      return this.context.text;
  }

  render() {
      return <div>{this.context.text}</div>;
  }
}

describe('your description', () => {
    it('your spec', () => {
        const context = { text: 'new value' };
        const ComponentWithContext = withContext(Component, context);

        const wrapper = shallow(<ComponentWithContext />, { context });

        expect(wrapper.instance().foo()).toBe('new value'); // fails as `.foo()` returns `undefined`
    });
});
```

**Comment by mjancarik** [comment]
It's should be fixed in shallow-with-context@0.2.0.

**Comment by connorjburton** [comment]
Thank you, really appreciate it!
